packages/flutter_tools/templates/app/lib/main.dart.tmpl

```diff
@@ -38,111 +38,182 @@ void main() {
 }
 
 {{^withPluginHook}}
+// This is a stateless widget that is the root widget for this application.
+//
+// It is said to be "stateless" because its attributes are fixed at creation, so
+// it can't keep any modified application state itself. The state it uses all
+// comes from its inputs. The power of stateless widgets is that although they
+// are lightweight and immutable, they can have logic in their build methods to
+// modify what gets built based on their configuration.
 class MyApp extends StatelessWidget {
   const MyApp({super.key});
 
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
+      title: 'Counter Example',
       theme: ThemeData(
-        // This is the theme of your application.
+        // This is the theme of the application.
+        //
+        // TRY THIS: Try running the application with "flutter run" on the
+        // command line, or the run button in the IDE. You'll see the
+        // application has a purple toolbar. Then, without quitting the app, try
+        // changing the colorScheme below to use a seedColor of Colors.green and
+        // then invoke "hot reload" (press "r" in the console where you ran
+        // "flutter run", save the changes to "hot reload" in a Flutter IDE, or
+        // press the "hot reload" button in a Flutter IDE).
         //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
         // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
+        // state is not lost during the reload. This is what is meant by
+        // "stateful hot reload". To reset the state, use hot restart instead.
+        //
+        // This works for code too, not just values: Most code changes can be
+        // tested with just a hot reload.
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+
+        // This makes the theme conform to a more up-to-date version of the
+        // Material Design specification. Soon, this will be the default, and
+        // this flag won't be necessary.
+        //
+        // See https://m3.material.io/ for more information about Material
+        // Design.
```

Comment:
if we add this, then "soon" is a word we'd will be seeing in examples derived from this for years.

I would probably just skip updating this template until we can make it the default. If it'll happen soon anyway, what's the rush. We have a few months until the next stable cut.

Reply:
Soon is relative.  It will happen sometime after the next stable cut.

I agree that adding a promise in the template is probably a bad idea, though.

Reply:
Sometimes after the next stable cut is definitely not soon. :-)